### PR TITLE
Mullinsmmm code review

### PR DESCRIPTION
**Pull request by PerryBaran**

### CODE REVIEW

File: src/kata1.fizzBuzz.js [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+//     if (number == 3){
+//      return ('Fizz');
+//     }; 
+//  };
+
+// const fizzBuzz = (number) => {
+//     if ((number / 3) == 0){
+//         return ('Fizz')
+//     };
+// };
```
Comment (PerryBaran) [inline src/kata1.fizzBuzz.js:22]: remove commented out code

File: src/kata1.fizzBuzz.js [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+const fizzBuzz = (number) => {
+    if(number % 3 === 0 && number % 5 === 0) {
+        return 'FizzBuzz';
+    } else if (number % 5 === 0) {
+        return 'Buzz';
+    } else if (number % 3 === 0) {
+        return 'Fizz';
+    } else {
+        return number;
+    }
```
Comment (PerryBaran) [inline src/kata1.fizzBuzz.js:33]: as your doing the same tests multiple times you can assign them to a variable to reduce code repetition, then call these constants in the `if` statements. i.e. add the variables `const isDivisibleBy3 = number % 3 === 0` and `const isDivisibleBy5 = number % 5 === 0`.

You can also just use `if` statements instead of `if else` statements, as if something is returned, nothing after the return will get called. i.e. refactor to
```
const isDivisibleBy3 = number % 3 === 0;
const isDivisibleBy5 = number % 5 === 0;

if (isDivisibleBy3 && isDivisibleBy5) return "FizzBuzz";
if (isDivisibleBy5) return "Buzz";
if (isDivisibleBy3) return "Fizz";
return number

```

File: src/kata2.booleanToWord.js [hunk 3]
```diff
@@ -1,3 +1,10 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
+    if(boolean == true) {
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:2]: Should always use strict equality in JavaScript i.e `===` instead of `==`. Although, in this case, this can be simplified by just writing `if(boolean)`

File: src/kata2.booleanToWord.js [hunk 4]
```diff
@@ -1,3 +1,10 @@
...
+        return ('Yes')
+    } else if (boolean == false) {
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:4]: unnecessary else if, if boolean isn't true, the boolean must be false

File: src/kata2.booleanToWord.js [hunk 5]
```diff
@@ -1,3 +1,10 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
+    if(boolean == true) {
+        return ('Yes')
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:3]: don't need parenthesis around the string

File: src/kata2.booleanToWord.js [hunk 6]
```diff
@@ -1,3 +1,10 @@
...
+    } else if (boolean == false) {
+        return ('No')
+    }
+};
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:7]: This function can be further simplified by using the ternary operator. e.g. `return boolean ? "Yes" : "No";`

File: src/kata3.numberToReversedDigits.js [hunk 7]
```diff
@@ -1,3 +1,40 @@
...
+//     const arr = [];
+//     arr.push(number);
+//     arr.join(',');
+
+// const arr = [];
+// for (let i = 0; i < number.length; i++) {
+//   arr.push(number[i]);
+// }
+// return arr.reverse().join(',');
+// };
```
Comment (PerryBaran) [inline src/kata3.numberToReversedDigits.js:29]: remove commented out code

File: src/kata3.numberToReversedDigits.js [hunk 8]
```diff
@@ -1,3 +1,40 @@
...
+
+    const digits = [];
+    while (number) {
+      digits.push(number % 10);
+      number = Math.floor(number / 10);
+    }
+    return digits;
+  };
```
Comment (PerryBaran) [inline src/kata3.numberToReversedDigits.js:37]: interesting approach to this problem, actually very similar to a coding challenge I did for a job interview

File: src/kata4.humanCatDogYears.js [hunk 9]
```diff
@@ -1,3 +1,35 @@
...
+//         years.push (24);
+//     } 
+//     else if (number => 3){
+//         years.push (24 + (number - 2)* 4, 24 + (number - 2) * 5);
+//     }
+//   }
+
+//   return years;
+// };
+
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:22]: remove comments and commented out code

File: src/kata4.humanCatDogYears.js [hunk 10]
```diff
@@ -1,3 +1,35 @@
...
+const humanCatDogYears = (number) => {
+    if (number === 1) {
+      return [number, 15, 15];
+    } 
+    if (number === 2) {
+      return [number, 24, 24];
+    } 
+    if (number >= 3) {
+      return [number, 24 + (number - 2) * 4, 24 + (number - 2) * 5];
+    }
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:32]: what if `number === 0` or `number < 0` ?

File: src/kata5.reachDestination.js [hunk 11]
```diff
@@ -1,3 +1,12 @@
-const reachDestination = (distance, speed) => {};
+// should return a string "I should be there in (time) hours."
+// 2 inputs (distance & speed) both of which are known
+// work out (time) by distance / speed.
+// (time) rounded to the nearest 0.5
```
Comment (PerryBaran) [inline src/kata5.reachDestination.js:4]: remove comments

File: src/kata5.reachDestination.js [hunk 12]
```diff
@@ -1,3 +1,12 @@
...
+
+const reachDestination = (distance, speed) => {
+  const time = Math.round((distance / speed) * 2)/2;
+
+  return "I should be there in " + time + " hours.";
```
Comment (PerryBaran) [inline src/kata5.reachDestination.js:9]: I would use a template literal for this, i.e. `return I should be there in ${time} hours.";`

Comment (PerryBaran) [reply src/kata5.reachDestination.js:9]: also, if `time === 1` you return `"... 1 hours."` instead of `"1 hour."`

File: src/kata6.joinNames.js [hunk 13]
```diff
@@ -1,3 +1,26 @@
...
+// array of objects needs to be for any amount of names
+
+//need to convert array of objects to string
+//then seperate with, 
+// create last name in string as a variable and put '&' 
+// between last names
+
+// const joinNames = (namesObj) => {
+//     const string = namesObj.name.split(",");
+// };
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:12]: remove comments

File: src/kata6.joinNames.js [hunk 14]
```diff
@@ -1,3 +1,26 @@
...
+
+const joinNames = (namesObj) => {
+    let string = '';
+    for (let i = 0; i < namesObj.length; i++) {
+      if (i === namesObj.length - 2) {
+        string += namesObj[i].name + ' & ' + namesObj[i+1].name;
+        break;
+      }
+      string += namesObj[i].name + ', ';
+    }
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:22]: instead of using `break`, could use `if(...) {...} else {...}`

File: test/kata1.fizzBuzz.test.js [hunk 15]
```diff
@@ -1,11 +1,17 @@
...
 
-  test("returns FizzBuzz when passed a multiple 3 and 5", () => {});
+  it("returns FizzBuzz when passed a multiple 3 and 5", () => {});
+  expect(fizzBuzz(15)).toBe('FizzBuzz')
+  expect(fizzBuzz(30)).toBe('FizzBuzz')
 
-  test("returns the number when it isn't a multiple of 3 or 5", () => {});
+  it("returns the number when it isn't a multiple of 3 or 5", () => {});
+  expect(fizzBuzz(4)).toBe(4)
+  expect(fizzBuzz(11)).toBe(11)
```
Comment (PerryBaran) [inline test/kata1.fizzBuzz.test.js:16]: Don't mix both single and double quotation marks in one project. In this project, eslint has been setup to use double quotation marks so should you.

File: test/kata2.booleanToWord.test.js [hunk 16]
```diff
@@ -1,5 +1,15 @@
+// When booleanToWord is passed true, 
+// it should return 'Yes'. When it is 
+// passed false, it should return 'No'.
```
Comment (PerryBaran) [inline test/kata2.booleanToWord.test.js:3]: remove comments

File: test/kata1.fizzBuzz.test.js (hunk 15 above)
Comment (PerryBaran) [inline test/kata1.fizzBuzz.test.js:16]: None of these tests are actually testing for anything, as your `expect` statement is outside of the callback statements passed to `it`. Currently, your doing
```
it("...", () => {});
expect(...).toBe(...);
```
when it should be written as
```
it("...", () => {
  expect(...).toBe(...);
});
```

File: test/kata2.booleanToWord.test.js [hunk 17]
```diff
@@ -1,5 +1,15 @@
...
 
 describe("booleanToWord", () => {
-  // how do we create specs again???
+  it("return Yes when passed true", () => {});
+  expect(booleanToWord(true)).toBe('Yes')
+});
+
+describe("booleanToWord", () => {
+  it("return No when passed false", () => {});
+  expect(booleanToWord(false)).toBe('No')
```
Comment (PerryBaran) [inline test/kata2.booleanToWord.test.js:14]: not testing anything as the `expect` statement is outside of the callback passed to `it`

File: test/kata2.booleanToWord.test.js [hunk 18]
```diff
@@ -1,5 +1,15 @@
...
 });
```
Comment (PerryBaran) [inline test/kata2.booleanToWord.test.js:15]: These tests should be in the same describe block, i.e.

```
describe("booleanToWord", () => {
  it("return Yes when passed true", () => {
    expect(booleanToWord(true)).toBe('Yes')
  });

  it("return No when passed false", () => {
    expect(booleanToWord(false)).toBe('No')
  });
});
```

File: test/kata3.numberToReversedDigits.test.js [hunk 19]
```diff
@@ -1,5 +1,6 @@
 const { numberToReversedDigits } = require("../src");
 
 describe("numberToReversedDigits", () => {
-  test("returns a reversed array of the number's digits", () => {});
+  it("returns a reversed array of the number's digits", () => {});
+  expect(numberToReversedDigits(4321)).toEqual([1,2,3,4])
```
Comment (PerryBaran) [inline test/kata3.numberToReversedDigits.test.js:5]: not testing anything as the `expect` statement is outside of the callback passed to `it`

File: test/kata4.humanCatDogYears.test.js [hunk 20]
```diff
@@ -1,3 +1,19 @@
...
+  describe("humanCatDogYears", () => {
+    it("for the second human year return 9 animal years", () => {});
+    expect(humanCatDogYears(2)).toEqual([2,24,24])
+  });
+
+  describe("humanCatDogYears", () => {
+    it("for every subsequent year add 4 cat years & 5 Dog years", () => {});
+    expect(humanCatDogYears(3)).toEqual([3,28,29])
+    expect(humanCatDogYears(10)).toEqual([10,56,64])
+  });
```
Comment (PerryBaran) [inline test/kata4.humanCatDogYears.test.js:17]: not testing anything as the `expect` statement is outside of the callback passed to `it`, and all tests should be inside of the same describe blocks

**Review by PerryBaran — COMMENTED** [review]
Hey Matt, I've left comments throughout for improvements, but here's a quick overview:
- need to remove unnecessary comments and commented out code
- linting issues - if you are getting red squiggly lines in your code editor make sure to fix them. If you hover your mouse over the red squiggly line it will tell you what the issue is. Usually, at the bottom of the suggestion box it will have a button called `Quick Fix... (Ctrl+.)` which you can click to quickly fix these issues.
- lots of the tests weren't actually testing for anything as the `expect` statement was outside of the callback passed to `it`. To test is a test is working properly, you can try and get the test to fail by asserting the wrong outcome e.g. `expect(booleanToWord(true)).toBe("sffsdgsrge")`, if the test still passes you know somethings gone wrong.
- after fixing your tests, they do still all pass, so all your solutions do work for the test cases provided, although you could add more tests for edge cases (e.g. if `time === 1` in `reachDestination` or if `number === 0` in `humanCatDogYears`)
